src/pages/index.js

```diff
@@ -4,73 +4,201 @@ import Section from "../components/Section.js";
+    constants.config,
+    document.querySelector("#avatar-form")
+  );
+
+  profileEditValidator.enableValidation();
+  addCardValidator.enableValidation();
+  avatarValidator.enableValidation();
+
+  const avatarImagePopup = new PopupWithForm("#edit-avatar-popup", (data) => {
+    avatarImagePopup.renderLoading(true);
+
+    api
+      .updateAvatar(data.link)
+      .then((res) => {
+        userInformation.setUserAvatar(res);
+      })
+      .catch((err) => {
+        console.error("Error updating avatar:", err);
+      })
+      .finally(() => {
+        avatarImagePopup.renderLoading(false);
+        avatarImagePopup.close();
+        avatarImagePopup.resetForm();
+        avatarValidator.toggleButtonState();
+      });
+  });
+
+  avatarImagePopup.setEventListeners();
+  document.querySelector(".profile__avatar").addEventListener("click", () => {
+    console.log("Avatar image clicked");
+    avatarValidator.resetValidation();
+    avatarValidator.toggleButtonState();
+    avatarImagePopup.open();
+  });
+
+  const profileEditForm = new PopupWithForm("#profile-edit-popup", (data) => {
+    const { header, description } = data;
+    profileEditForm.renderLoading(true);
 
-cardSection.renderItems(constants.initialCards);
+    api
+      .updateUserInfo(header, description)
+      .then((res) => {
+        userInformation.setUserInfo(res);
+      })
+      .catch((err) => {
+        console.error("Error updating profile:", err);
+      })
+      .finally(() => {
+        profileEditForm.renderLoading(false);
+        profileEditForm.close();
+        profileEditForm.resetForm();
+        profileEditValidator.toggleButtonState();
+      });
+  });
+
+  constants.profileEditButton.addEventListener("click", () => {
+    profileEditValidator.resetValidation();
+    profileEditValidator.toggleButtonState();
+    const userInfo = userInformation.getUserInfo();
+    constants.profileTitleInput.value = userInfo.title || "";
+    constants.profileDescriptionInput.value = userInfo.description || "";
+    profileEditForm.open();
+  });
+
+  profileEditForm.setEventListeners();
+
+  function handleImageClick(name, link) {
+    cardPreview.open({ name, link });
+  }
+
+  function createCard(cardData) {
+    const cardElement = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    );
+    return cardElement.getView();
+  }
+
+  const addCardForm = new PopupWithForm("#add-card-popup", (data) => {
+    const { title, URL } = data;
+    addCardForm.renderLoading(true);
+
+    api
+      .addCard(title, URL)
+      .then((cardData) => {
+        cardSection.addItem(createCard(cardData));
+      })
+      .catch((err) => {
+        console.error("Error adding card:", err);
+      })
+      .finally(() => {
+        addCardForm.renderLoading(false);
+        addCardForm.close();
+        addCardForm.resetForm();
+        addCardValidator.toggleButtonState();
```

Comment:
* You shouldn't have placed closing and resetting in `finally`. You need to add `toggleButtonState` only on **opening** to toggle the button